### PR TITLE
DAOS-623 build: Remove client/server libs from common package

### PR DESCRIPTION
**Pull request by jolivier23**
Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

File: src/SConscript [hunk 1]
```diff
@@ -12,9 +12,6 @@ HEADERS_GURT = ['dlog.h', 'debug.h', 'common.h', 'hash.h', 'list.h',
                 'heap.h', 'fault_inject.h', 'debug_setup.h',
                 'types.h', 'atomic.h', 'slab.h']
 
-# Keep versioned libs for now to avoid any conflict with 1.0
-CART_VERSION = "4.9.0"
```
Comment (brianjmurrell) [inline src/SConscript:16]: I don't think you can do this.  (Right or wrong) there are consumers of libgurt outside of DAOS:

Problem 1: package ior-hpc-3.3.0-13.633.g89fd198.el7.x86_64 requires libgurt.so.4()(64bit), but none of the providers can be installed

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: src/SConscript (hunk 1 above)
Comment (jolivier23) [reply src/SConscript:16]: ok, added version back for gurt

Comment (jolivier23) [reply src/SConscript]: mpich seems to want cart.  Not sure why. I'm pretty sure it's an indirect dependence

**Comment by jolivier23** [comment]
@brianjmurrell I think this one can work now

**Review by frostedcmos — APPROVED** [review]

**Review by mjmac — APPROVED** [review]

File: debian/daos.install [hunk 2]
```diff
@@ -1,8 +1,7 @@
-usr/lib64/libdaos_common.so
 etc/daos/memcheck-cart.supp
-# TODO: this should move from daos_srv to daos
```
Comment (brianjmurrell) [inline debian/daos.install:3]: Should this TODO have moved also, or is this no longer a TODO?

File: utils/rpms/daos.spec [hunk 3]
```diff
@@ -338,14 +338,12 @@ getent passwd daos_agent >/dev/null || useradd -s /sbin/nologin -r -g daos_agent
 %dir %{conf_dir}
 %dir %{_sysconfdir}/bash_completion.d
 %{_sysconfdir}/bash_completion.d/daos.bash
-%{_libdir}/libdaos_common.so
-%{_libdir}/*.so.*
-# TODO: this should move from daos_srv to daos
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec:343]: Ditto.

**Review by brianjmurrell — APPROVED** [review]

File: debian/daos.install (hunk 2 above)
Comment (jolivier23) [reply debian/daos.install:3]: It's no longer a todo.

**Pull request merged**